### PR TITLE
Add `d/aws_identitystore_groups`

### LINKED ISSUE #26770: New Datasources: aws_identitystore_users & aws_identitystore_groups (plural)
Fixes https://github.com/hashicorp/terraform-provider-aws/issues/17279

> I'd appreciate both singular and plural data sources (e.g. `aws_identitystore_group` and `aws_identitystore_groups`), [...].

_Originally posted by @lorengordon in https://github.com/hashicorp/terraform-provider-aws/issues/17279#issuecomment-1030210907_

`aws_identitystore_group` & `aws_identitystore_user` were made to fetch a single user or group. It would be nice to have the same resources but plural. These would return a list (with 0, 1 or many elements).

Current workaround:
```hcl
locals {
  aws_sso_users = {
    # aws identitystore list-users --identity-store-id d-xxxxxxxxxx --query 'Users[].[UserName,UserId]' --output text | awk '{print "\""$1"\" = \""$2"\","}'
    "toto@example.com"           = "1234567891-90825474-0913-xxxx-wwww-abdc5783ff98",
    "jojo@example.com"           = "1234567891-42349857-7836-yyyy-zzzz-192837465302",
    # ...
  }
}
```

Proposed solution:
```hcl
data "aws_identitystore_users" "every_user" {
  identity_store_id = tolist(data.aws_ssoadmin_instances.example.identity_store_ids)[0]
}
```

### PR DESCRIPTION
**Pull request by aristosvo**
### Description
Follow-up of #31681, big thanks for @liath laying the foundation for this one.

Updated to plugin framework
>  Adds a data source for listing all groups in an Identity Store instance. Part of a few PRs to cover https://github.com/hashicorp/terraform-provider-aws/issues/26770.
> 
> The acceptance test is weird and I'd appreciate guidance on what to do there. The issue is that Identity Store instances are account wide so we can't guarantee there's only one group in the results or the order of the results. As far as I could find there isn't a way to say in the AttrPair checks "check these nested fields in any item of this list and compare those to these attributes on this other resource". So we're left with doing that in a custom test function.
> 
> Of course, server-side filter would likely help here but aws-sdk says that functionality is deprecated. :(

The importance of this for us a customer is mainly to automate SSO assignments within our landing zone and account vending, otherwise using scripts covering this missing functionality.

### Relations
Followup of #31681 
Partly closes #26770

### References
@bschaatsbergen wanted to help me move this forward

### Output from Acceptance Testing

```console
% make testacc TESTS=TestAccIdentityStoreGroupsDataSource_ PKG=identitystore  
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.21.8 test ./internal/service/identitystore/... -v -count 1 -parallel 20 -run='TestAccIdentityStoreGroupsDataSource_'  -timeout 360m
=== RUN   TestAccIdentityStoreGroupsDataSource_basic
=== PAUSE TestAccIdentityStoreGroupsDataSource_basic
=== CONT  TestAccIdentityStoreGroupsDataSource_basic
--- PASS: TestAccIdentityStoreGroupsDataSource_basic (12.22s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/identitystore      16.670s
```

### CODE REVIEW

File: internal/service/identitystore/groups_data_source_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+			acctest.PreCheckSSOAdminInstances(ctx, t)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, names.IdentityStoreServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		// All destroy checks are already covered in sibling tests
+		CheckDestroy: acctest.CheckDestroyNoop,
```
Comment (bschaatsbergen) [inline internal/service/identitystore/groups_data_source_test.go:36]: I couldn't find the implementing of CheckDestroy, is it covered somewhere?

File: website/docs/d/identitystore_groups.html.markdown [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## Example Usage
+
+### Basic Usage
+
+```terraform
+data "aws_ssoadmin_instances" "example" {}
+
+data "aws_identitystore_groups" "example" {
+  identity_store_id = tolist(data.aws_ssoadmin_instances.example.identity_store_ids)[0]
```
Comment (bschaatsbergen) [inline website/docs/d/identitystore_groups.html.markdown:21]: Was there a particular reason this is converted to a list? I'm curious if this works:

```suggestion
  identity_store_id = data.aws_ssoadmin_instances.example.identity_store_ids[0]
```

File: website/docs/d/identitystore_groups.html.markdown [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are required:
+
+* `identity_store_id` - (Required) Identity Store ID associated with the Single Sign-On Instance.
```
Comment (bschaatsbergen) [inline website/docs/d/identitystore_groups.html.markdown:29]:
```suggestion
* `identity_store_id` - (Required) Identity Store ID associated with the Single Sign-On (SSO) Instance.
```

**Review by bschaatsbergen — COMMENTED** [review]
Hey @aristosvo, great work! Thanks a lot for picking this up. The PR looks really good.

Let me check with the team to see if we have an AWS account with an SSO instance set up already. Meanwhile I've left a few comments.

File: website/docs/d/identitystore_groups.html.markdown (hunk 2 above)
Comment (aristosvo) [reply website/docs/d/identitystore_groups.html.markdown:21]: This is incorporated in the tests and probably based on the documentation of the `aws_ssoadmin_instances` docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ssoadmin_instances.

Would not have done it myself this way as well, inherited from the original PR. I'll check whether we can remove the `tolist` function

File: internal/service/identitystore/groups_data_source_test.go (hunk 1 above)
Comment (aristosvo) [reply internal/service/identitystore/groups_data_source_test.go:36]: Will look into it!

File: website/docs/d/identitystore_groups.html.markdown (hunk 2 above)
Comment (bschaatsbergen) [reply website/docs/d/identitystore_groups.html.markdown:21]: If removing the `tolist` works, I'll create a follow-up PR to address that example in the docs 👍🏼

Comment (aristosvo) [reply website/docs/d/identitystore_groups.html.markdown]: Validated, removed ✅

Comment (bschaatsbergen) [reply website/docs/d/identitystore_groups.html.markdown]: Awesome

File: internal/service/identitystore/groups_data_source_test.go (hunk 1 above)
Comment (aristosvo) [reply internal/service/identitystore/groups_data_source_test.go]: I've used the CheckDestroy now which is used in the `group_data_source` tests, seems to cover the functionality needed

**Review by bschaatsbergen — APPROVED** [review]
LGTM 🚀 - looped in the AWS team internally to run the tests against an AWS account with SSO instances configured.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccIdentityStoreGroupsDataSource_' PKG=identitystore
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.21.8 test ./internal/service/identitystore/... -v -count 1 -parallel 20  -run=TestAccIdentityStoreGroupsDataSource_ -timeout 360m
=== RUN   TestAccIdentityStoreGroupsDataSource_basic
=== PAUSE TestAccIdentityStoreGroupsDataSource_basic
=== CONT  TestAccIdentityStoreGroupsDataSource_basic
--- PASS: TestAccIdentityStoreGroupsDataSource_basic (18.14s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/identitystore	29.616s
```

**Comment by ewbankkit** [comment]
@aristosvo Thanks for the contribution :tada: :clap:.

**Comment by bschaatsbergen** [comment]
Thanks for your contributions @aristosvo & @liath 🎉👏

**Comment by aristosvo** [comment]
Thank you @bschaatsbergen for helping us push it to the finish line 🏋️🏁

**Pull request merged**